bin/proving-service/src/proxy/worker.rs

```diff
@@ -32,7 +32,7 @@ const MPS_PROXY_VERSION: &str = env!("CARGO_PKG_VERSION");
 #[derive(Debug, Clone)]
 pub struct Worker {
     backend: Backend,
-    status_client: StatusApiClient<Channel>,
+    status_client: Option<StatusApiClient<Channel>>,
     is_available: bool,
     health_status: WorkerHealthStatus,
     version: String,
```

Comment:
Not from this PR, but it would be good to add comments for the fields to explain what they are - especially for `is_available` and `health_status` fields (and the description on lines 30 - 31 also needs to be updated).